packages/flutter/lib/src/rendering/debug_overflow_indicator.dart

```diff
@@ -0,0 +1,301 @@
+        rect: markerRect,
+        label: 'BOTTOM OVERFLOWED BY ${_formatPixels(overflow.bottom)} PIXELS',
+        labelOffset: markerRect.bottomCenter -
+            const Offset(0.0, _indicatorFontSizePixels + _indicatorLabelPaddingPixels),
+        rotation: 0.0,
+      ));
+    }
+    return regions;
+  }
+
+  void _reportOverflow(RelativeRect overflow, String overflowHints) {
+    overflowHints ??= 'The edge of the $runtimeType that is '
+      'overflowing has been marked in the rendering with a yellow and black '
+      'striped pattern. This is usually caused by the contents being too big '
+      'for the $runtimeType.\n'
+      'This is considered an error condition because it indicates that there '
+      'is content that cannot be seen. If the content is legitimately bigger '
+      'than the available space, consider clipping it with a ClipRect widget '
+      'before putting it in the $runtimeType, or using a scrollable '
+      'container, like a ListView.';
+
+    final List<String> overflows = <String>[];
+    if (overflow.left > 0.0)
+      overflows.add('${_formatPixels(overflow.left)} pixels on the left');
+    if (overflow.top > 0.0)
+      overflows.add('${_formatPixels(overflow.top)} pixels on the top');
+    if (overflow.bottom > 0.0)
+      overflows.add('${_formatPixels(overflow.bottom)} pixels on the bottom');
+    if (overflow.right > 0.0)
+      overflows.add('${_formatPixels(overflow.right)} pixels on the right');
+    String overflowText = '';
+    assert(overflows.isNotEmpty,
+        "Somehow $runtimeType didn't actually overflow like it thought it did.");
+    switch (overflows.length) {
+      case 1:
+        overflowText = overflows.first;
+        break;
+      case 2:
+        overflowText = '${overflows.first} and ${overflows.last}';
+        break;
+      default:
+        overflows[overflows.length - 1] = 'and ${overflows[overflows.length - 1]}';
+        overflowText = overflows.join(', ');
+    }
+    FlutterError.reportError(
+      new FlutterErrorDetailsForRendering(
+        exception: 'A $runtimeType overflowed by $overflowText.',
+        library: 'rendering library',
+        context: 'during layout',
+        renderObject: this,
+        informationCollector: (StringBuffer information) {
+          information.writeln(overflowHints);
+          information.writeln('The specific $runtimeType in question is:');
+          information.writeln('  ${toStringShallow(joiner: '\n  ')}');
+          information.writeln('◢◤' * (FlutterError.wrapWidth ~/ 2));
+        },
+      ),
+    );
+  }
+
+  /// To be called when the overflow indicators should be painted.
+  ///
+  /// Typically only called if there is an overflow, and only from within a
+  /// debug build.
+  ///
+  /// See example code in [DebugOverflowIndicatorMixin] documentation.
+  void paintOverflowIndicator(
+      PaintingContext context, Offset offset, Rect containerRect, Rect childRect,
+      {String overflowHints}) {
```

Comment:
i think if you put a comma after "overflowHints", dartfmt will give you a better wrapping here